FluentTerminal.App.ViewModels/TerminalViewModel.cs

```diff
@@ -341,11 +341,22 @@ private async void OnKeyboardCommandReceived(object sender, string command)
                 var selection = await _terminalView.GetSelection().ConfigureAwait(true);
                 _clipboardService.SetText(selection);
             }
-            else if (command == nameof(Command.Paste))
+            else if ((command == nameof(Command.Paste)) || (command == nameof(Command.PasteWithoutNewlines)))
             {
-                var content = await _clipboardService.GetText().ConfigureAwait(true);
+                string content = await _clipboardService.GetText().ConfigureAwait(true);
                 if (content != null)
                 {
+                    // Use the shell profile line-ending translation method to translate the line endings of the pasted buffer
+                    // or delete all of the instances of the newline pattern by replacing with the empty string.
+                    if (command == nameof(Command.Paste))
+                    {
+                        content = _shellProfile.TranslateLineEndings(content);
+                    }
+                    else
```

Comment:
Perhaps make more explicit for clarity (don't know if the maintainers are a fan of this though)?

```suggestion
                    else if (command == nameof(Command.PasteWithoutNewlines))
```

Reply:
Resolved by converting that section to a `switch() { }`.

Reply:
Great solution 👍

Stylistically I'm unsure whether the braces in `case: { }` are desired, I've personally never seen it and as far as I can tell it's not used anywhere else in the project. That said, don't feel the need to change it unless the maintainer(s) say something about it 😄

Reply:
Stylistically, that's a hold over from ANSI style bracing of C++, and is (at least, in C++), required if you ever define a local scope variable inside a switch case like we do for the two Paste commands). I'm not married to it, but trained myself over years of using `astyle -j --style ansi`.

Reply:
Makes sense, didn't know that. Thanks for the explanation 🙂